solarforecastarbiter/metrics/preprocessing.py

```diff
@@ -58,6 +58,116 @@ def apply_validation(obs_df, qfilter, handle_func):
         return validated_obs, counts
 
 
+def _resample_event_obs(obs, fx, obs_series):
+    """
+    Resample the event observation.
+
+    Parameters
+    ----------
+    obs : datamodel.Observation
+        The Observation being resampled.
+    fx : datamodel.EventForecast
+        The corresponding Forecast.
+    obs_series : pd.Series
+        Timeseries data of the event observation.
+
+    Returns
+    -------
+    obs_resampled : pandas.Series
+        Timeseries data of the Observation resampled to match the Forecast.
+
+    Raises
+    ------
+    RuntimeError
+        If the Forecast and Observation do not have the same interval length.
+
+    """
+
+    if fx.interval_length != obs.interval_length:
+        raise ValueError("Event observation and forecast time-series "
+                         "must have matching interval length.")
+    else:
+        obs_resampled = obs_series
+
+    return obs_resampled
+
+
+def _validate_event_dtype(ser):
+    """
+    Validate the event data dtype, converting to boolean values if possible.
+
+    Parameter
+    ---------
+    ser : pandas.Series
+        The event time-series data (observation or forecast).
+
+    Returns
+    -------
+    ser : pandas.Series
+        The event time-series data as boolean values.
+
+    Raises
+    ------
+    TypeError
+        If the event time-series data dtype cannot be converted to boolean.
+
+    """
+
+    if ser.dtype == bool:
+        return ser
+    elif ser.dtype == int and np.all(np.isin(ser.unique(), [0, 1])):
+        return ser.astype(bool)
+    elif ser.dtype == float and np.all(np.isin(ser.unique(), [0.0, 1.0])):
+        return ser.astype(int).astype(bool)
```

Comment:
arrays of floats can be cast to bools, so I don't understand why we go through ints. If it's needed, add a comment.

Reply:
I ran into an issue previously where casting from floats to bools directly threw an error, but I can't reproduce that error now. So maybe it was something else.

Anyways, I'll revise to cast directly to bool from float.